### DISCUSSION TITLE
The True ROI of Doing It Right the First Time in Software Projects

### DISCUSSION THREAD

**Discussion by Bisaloo** [epiverse-trace/epiverse-trace.github.io#267] [Resources]
### Context

There is a common misunderstanding that agile should mean "prioritize quick and dirty solutions".

This blog post presents an alternative view, with the focus on carefully defining a **really minimal** MVP, which can be done right from the first iteration, without spending too much time.

### URL

https://www.appsilon.com/post/do-it-right-the-first-time

### Suggested labels

Best practices, Agile, Development Process

**Comment by adamkucharski** [comment]
Thanks for sharing. One aspect of that piece that seems particularly pertinent to epidemic response projects is the prototype-to-MVP step, e.g. some code is developed as a quick exploration or sense-check, but then ends up being reused far more than planned. Tagging @hillalex as had some related discussions about where early stage work should sit recently. And related to [this discussion on sharing project code](https://github.com/orgs/epiverse-trace/discussions/186) and what a minimum package should be.
